### PR TITLE
[`SAM`] fix sam slow test

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR fixes the slow tests that were failing due to https://github.com/huggingface/transformers/pull/23295

In fact, in the slow test that we have designed, we forgot to use the correct format for the input bounding boxes

Will address a PR on `notebooks` to address the changes in the example notebook

cc @ydshieh @amyeroberts

### CODE REVIEW

**Comment by younesbelkada** [comment]
Fix for the notebook: https://github.com/huggingface/notebooks/pull/371

**Review by ydshieh — APPROVED** [review]
Thanks!

**Review by amyeroberts — COMMENTED** [review]
Thanks for fixing! Good to see type of boxes and points now consistent. 

Before merging, so that I understand the consequences - how did the slow tests run before? As I understand it, the tests are now failing because of the assertions being raised based on input type checks. However, the previous inputs were still being successfully processed i.e. the test would run without issue and produce the same results. If this is the case - do we need the logic checks from #23295? 

As an additional side note, I just noticed the [exception messages](https://github.com/huggingface/transformers/blob/ee3be05310d59c870a930c841eb122694bfdf5b5/src/transformers/models/sam/processing_sam.py#L211) should be updated - you can't have floating integers :)

File: src/transformers/models/sam/processing_sam.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ def _check_and_preprocess_points(
                 input_points = input_points.numpy().tolist()
 
             if not isinstance(input_points, list) or not isinstance(input_points[0], list):
-                raise ValueError("Input points must be a list of list of floating integers.")
+                raise ValueError("Input points must be a list of list of floating points.")
```
Comment (amyeroberts) [inline src/transformers/models/sam/processing_sam.py:211]: The other assertion message for `input_boxes` needs to be updated too :)

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing and taking the time to explain about previously passing tests!

**Comment by younesbelkada** [comment]
Thanks a lot @amyeroberts !

**Pull request merged**